### PR TITLE
[ext] ExtDefaultContextLoader & ExtDefaultMessageWriter

### PR DESCRIPTION
**Pull request by alangenfeld**
update the default context injector to handle direct from env var or file loading

update the default message writer to support outputting to standard streams or a file 

## How I Tested These Changes

existing tests

### CODE REVIEW

**Comment by alangenfeld** [comment]
Current dependencies on/for this PR:
* master
  * **PR #16318** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/16318" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/16318?utm_source=stack-comment).

File: python_modules/dagster-ext/dagster_ext/_io/default.py [hunk 1]
```diff
@@ -45,6 +67,14 @@ def write_message(self, message: ExtMessage) -> None:
             f.write(json.dumps(message) + "\n")
 
 
+class ExtStreamMessageChannel(ExtMessageWriterChannel):
+    def __init__(self, stream: TextIO):
+        self._stream = stream
+
+    def write_message(self, message: ExtMessage) -> None:
+        self._stream.writelines((json.dumps(message), "\n"))
```
Comment (alangenfeld) [inline python_modules/dagster-ext/dagster_ext/_io/default.py:75]: testing this in the existing suite would involve modifications to  `ExtSubprocess` in the same vein as highlighted here in k8s https://github.com/dagster-io/dagster/pull/16122/files#r1313558644

The crux of the problem here is needing to interact with post launch information in order to do reads.

File: python_modules/dagster-ext/dagster_ext/_io/default.py [hunk 2]
```diff
@@ -16,24 +17,45 @@
...
+            elif stream == "out":
+                yield ExtStreamMessageChannel(sys.stdout)
+            else:
+                raise DagsterExtError(
+                    f'Invalid value for key "std", expected stderr or stdout but received {stream}'
+                )
+        else:
+            raise DagsterExtError(
+                f'Invalid params for {self.__class__.__name__}, expected key "path" or "std",'
+                f" received {params}"
```
Comment (schrockn) [inline python_modules/dagster-ext/dagster_ext/_io/default.py:57]: Can we encode the magic strings "path" "std" etc somewhere so it is clear what is going on and so we can match it  to code on the launching side?

**Review by schrockn — CHANGES_REQUESTED** [review]
Less magic strings please.

File: python_modules/dagster-ext/dagster_ext/_io/default.py [hunk 3]
```diff
@@ -16,24 +17,45 @@
...
     @contextmanager
-    def open(self, params: ExtParams) -> Iterator["ExtFileMessageChannel"]:
-        path = assert_env_param_type(params, "path", str, self.__class__)
-        yield ExtFileMessageChannel(path)
+    def open(self, params: ExtParams) -> Iterator[ExtMessageWriterChannel]:
+        if "path" in params:
+            path = assert_env_param_type(params, "path", str, self.__class__)
+            yield ExtFileMessageChannel(path)
+        elif "std" in params:
+            stream = assert_env_param_type(params, "std", str, self.__class__)
```
Comment (smackesey) [inline python_modules/dagster-ext/dagster_ext/_io/default.py:45]: feel like this should be `"stdio"` instead of `std`

**Review by smackesey — APPROVED** [review]
One comment but otherwise good. I think we can address testing with possible changes to subprocess in a followup.

**Review by schrockn — APPROVED** [review]

**Pull request merged**